fix(lsp): open_floating_preview() ignores max_height

rmuir opened the pull request:
Problem:  After 47aaddfa the max_height option is no longer respected.
          Hover documentation and Signature help windows take up the
          entire text height.
Solution: Compare to window's current height and only modify the height
          if it would reduce the height, not enlarge it.

rmuir commented:
For example, currently this configuration:
```lua
vim.keymap.set('n', 'K', function()
  vim.lsp.buf.hover({ max_height = 10 })
end)
```
will result in hover doc taking up the entire screen :)
![Screen_Shot_2025-03-03_at_19 00 08](https://github.com/user-attachments/assets/f4f81369-d5d5-4139-b95f-c2bc33bc7028)

zeertzjq commented:
I don't think this is the current solution. It's better to compare `conceal_height` with `nvim_win_get_height` instead.

rmuir commented:
Thank you for looking, I reworked it to compare to window size and only reduce the window height (not enlarge it)

zeertzjq commented on runtime/lua/vim/lsp/util.lua:
```diff
@@ -1653,9 +1653,9 @@ function M.open_floating_preview(contents, syntax, opts)
     vim.wo[floating_winnr].concealcursor = 'n'
     vim.bo[floating_bufnr].filetype = 'markdown'
     vim.treesitter.start(floating_bufnr)
-    if not opts.height then
```
This condition is still needed. If `opts.height` is specified then that's the exact height of the floating window that shouldn't be reduced.

rmuir committed:
fix(lsp): open_floating_preview() ignores max_height

Problem:  After 47aaddfa the max_height option is no longer respected.
          Hover documentation and Signature help windows take up the
          entire text height.
Solution: Compare to window's current height and only modify the height
          if it would reduce the height, not enlarge it.

rmuir commented:
Thanks again, hopefully closer now: existing `height` check is still respected.
